Sources/AblyChat/Room.swift

```diff
@@ -20,18 +20,37 @@ public protocol Room: AnyObject, Sendable {
 internal actor DefaultRoom: Room {
     internal nonisolated let roomID: String
     internal nonisolated let options: RoomOptions
+    private let chatAPI: ChatAPI
+    
+    private let _messages: any Messages
 
     // Exposed for testing.
     internal nonisolated let realtime: RealtimeClient
 
-    internal init(realtime: RealtimeClient, roomID: String, options: RoomOptions) {
+    internal init(realtime: RealtimeClient, chatAPI: ChatAPI, roomID: String, options: RoomOptions) async {
         self.realtime = realtime
         self.roomID = roomID
         self.options = options
+        self.chatAPI = chatAPI
+        
+        self._messages = await DefaultMessages(
```

Comment:
Why `await`?

Reply:
@maratal for some reason I can't reply to https://github.com/ably-labs/ably-chat-swift/pull/45#discussion_r1780079477 so I’ll do it here ("I agree that creating initializers that must be called with `await` should be avoided.")

I wasn’t saying that it should necessarily be avoided (e.g. in #53 I’m making the room lifecycle manager initializer `async` because it needs to perform some `async` setup work before it can be used); I was just curious what the need was here.

Reply:
removed the need for await here but added a comment to the child class it's calling out to (the comment you linked to)